packages/core/crates/core-types/src/channels.rs

```diff
@@ -232,6 +232,84 @@ pub fn generate_channel_id(source: &Address, destination: &Address) -> Hash {
     Hash::create(&[&source.to_bytes(), &destination.to_bytes()])
 }
 
+/// Enumerates possible changes on a channel entry update
+#[derive(Clone, Copy, Debug)]
+pub enum ChannelChange {
+    /// Channel status has changed
+    Status { left: ChannelStatus, right: ChannelStatus },
```

Comment:
Having both directions is not OCP friendly. There might be strategies in the future, if a channel is open from A -> B, it does not have to be open from B -> A. So maybe the channel changes should observe each direction separately.

Reply:
They do actually! https://github.com/hoprnet/hoprnet/blob/29b6a61fd292bdf50218d7cf03baecdae03901da/packages/core/crates/core-strategy/src/strategy.rs#L37-L44